src/main/java/es/unizar/smartcampuz/model/reservation/TimeReservation.java

```diff
@@ -0,0 +1,73 @@
+package es.unizar.smartcampuz.model.reservation;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+@Embeddable
+public class TimeReservation {
+
+    @Transient
+    private static final int NUM_TIME_SLOTS = 24;
+
+    @NotNull
+    private int timeSlots;
+
+    //Used by Hibernate. Private is more readable. Protected will improve performance.
+    protected TimeReservation(){
+
+    }
+
+    public TimeReservation(@NotNull boolean[] timeSlots){
+        this.timeSlots=arrayToInteger(timeSlots);
+    }
+
+    //Used by Hibernate. Private is more readable. Protected will improve performance.
+    protected void setTimeSlots(boolean[] timeSlots) {
+        this.timeSlots = arrayToInteger(timeSlots);
+    }
+
+    public boolean[] getTimeSlots() {
+        return integerToArray(timeSlots);
+    }
+
+    private boolean[] integerToArray( int timeSlots ){
+        if( timeSlots >= (1<<24) ){
+            throw new IllegalArgumentException("Input should be higher than 1<<24");
```

Comment:
This kind of message prompts into admin terminal. I'm not sure if it's transmitted to the user. BTW, this line is "almost impossible to reach". This is defensive programming.